Jenkinsfile

```diff
@@ -5,17 +5,40 @@ elifeLibrary {
         commit = elifeGitRevision()
     }
 
-    stage 'Build image', {
-        sh 'docker build -t elife/sciencebeam-utils .'
-    }
+    node('containers-jenkins-plugin') {
+        def candidateVersion
+        stage 'Build images', {
+            checkout scm
+            dockerComposeBuild(commit)
+            try {
+                candidateVersion = sh(script: "IMAGE_TAG=${commit} docker-compose run --rm ./print_version.sh", returnStdout: true).trim()
+                echo "Candidate version: v${candidateVersion}"
+            } finally {
+                sh 'docker-compose down -v'
+            }
+        }
+
+        stage 'Project tests', {
+            try {
+                sh "IMAGE_TAG=${commit} docker-compose run --rm sciencebeam-utils ./project_tests.sh"
+            } finally {
+                sh 'docker-compose down -v'
+            }
+        }
+
+        stage 'Push release', {
+            // def isNew = sh(script: "git tag | grep v${candidateVersion}", returnStatus: true) != 0
```

Comment:
should be wrapped in https://github.com/search?q=org%3Aelifesciences+elifeMainlineOnly&type=Code to avoid executing in PRs